app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -74,6 +75,11 @@ function* destroyPageSession({ page, currentPageCount }) {
 function* naviatePageTo({ pageId, webview, location }) {
   yield put(PageActions.resetPageData(pageId));
   yield put(PageActions.setPageState(pageId, { load: PageState.STATES.PRE_LOADING }));
+  // Optimistically set page location and update the urlbar before the actual
+  // navigation happens, so that the UI shows the intended location instead of
+  // just some blank text.
+  yield put(PageActions.setPageDetails(pageId, { location }));
```

Comment:
These two actions seem to always be used together at least in the current 3 usages (including this one).  Is it possible / desirable to make the ui saga always run this setURLBarValue logic when the setPageDetails action happens with a location property?  I'm not sure if there's a time we'd want to setPageDetails to change the location *without* also updating the associated URL bar, and it would be an easy mistake to make